lib/index.js

```diff
@@ -594,7 +594,14 @@ function initialDepsNotMet(stream) {
  */
 function updateStream(s) {
   if ((s.depsMet !== true && initialDepsNotMet(s)) ||
-    (s.end !== undefined && s.end.val === true)) return;
+    (s.end !== undefined && s.end.val === true)) {
+    if (toUpdate.length > 0 && inStream !== undefined) {
```

Comment:
Not sure this is the correct check to make.

If we look at the `updateStreamValue` method we call `markListeners` when this situation occurs.

This means that streams which should be updated here will pass this check:

```
inStream !== undefined && s.deps.every(dep => dep.shouldUpdate)
```

I think the above check might provide better results as you're not updating everything, only things whose dependencies haven't been updated yet

Reply:
I'm wrong on this, the markListeners happens after you do this check :/

Reply:
That check I added does give another chance for streams to update when their dependencies are still missing values because they have been queued earlier because in their `updateStream` call `inStream` has been true because we're inside another stream. I think a better fix would be to use something else than a singular `inStream` value for handling nested streams but I haven't been able to come up with anything yet. Until a better solution is implemented I think a quick fix like this is reasonable.